Assets/Scripts/Player/WeaponInventory.cs

```diff
@@ -134,27 +148,42 @@ private void ChangeWeapon(int newWeaponIndex, bool force = false)
         {
             return;
         }
+        else if (playerGun.reloading)
+        {
+            return;
+        }
 
         Assert.Boolean(newWeaponIndex < Weapons.Count);
 
         Gun gun = GetComponent<Gun>();
 
-        if (currentWeaponIndex != -1)
+        if (currentWeaponIndex != -1 && Weapons.Count > currentWeaponIndex && !removed)

```

Comment:
Why do we need to check that `removed` is `false`?

Reply:
Stated above